### ISSUE TITLE
Contrast issues of button with pseudo element styles

### ISSUE THREAD

**Issue by davodavodavo3** [GoogleChrome/lighthouse#12584]
#### Provide the steps to reproduce
1. Run LH  Accessibility audit on https://theskinexperts.co.uk
2. Check Contrast issues related to the black button with white text

#### What is the current behavior?
Background and foreground colors do not have a sufficient contrast ratio. 
Element has insufficient color contrast of 1.07

LH and Axe are checking directly the button styles where background color is transparent, and they says that contrast of white text is not enough.
![image](https://user-images.githubusercontent.com/10462563/120087746-9ab7e780-c0fb-11eb-95d9-f190916e8b73.png)

#### What is the expected behavior?
LH/Axe should recognize that there is a :before pseudo element with background color of `#000000`  and in this case the button text contrast is enough 21.

#### Environment Information
* Affected Channels: DevTools
* Axe version: 4.1.2
* Lighthouse version: 7.2.0
* Chrome version: 91.0.4472.77
* Node.js version:
* Operating System: Windows 10

**Comment by patrickhulce** [comment]
Thanks for filing @davodavodavo3!

> Before creating an Accessibility issue please test that it is reproducible upstream with axe (https://www.deque.com/axe/) first and file the issue there if necessary.

Was this step already completed? I'm able to reproduce this with axe alone and Lighthouse not involved.

**Comment by davodavodavo3** [comment]
Hi @patrickhulce , yes Axe is showing the same.

**Comment by patrickhulce** [comment]
Sorry for miscommunication @davodavodavo3 I know axe is showing the same. I'm asking you if you've already filed a bug with axe per the [bug report template instructions](https://raw.githubusercontent.com/GoogleChrome/lighthouse/master/.github/ISSUE_TEMPLATE/Bug_report.md).

**Comment by davodavodavo3** [comment]
No, I didn't

**Comment by patrickhulce** [comment]
OK, would you be willing to? :)

**Comment by davodavodavo3** [comment]
Sorry @patrickhulce where to write?

Is this correct place [https://github.com/dequelabs/axe-core](https://github.com/dequelabs/axe-core)

**Comment by patrickhulce** [comment]
@davodavodavo3  Yep!

**Comment by davodavodavo3** [comment]
Already done. Thanks.

**Comment by davodavodavo3** [comment]
Two answers from Axe contributors.
https://github.com/dequelabs/axe-core/issues/2966#issuecomment-852012332
https://github.com/dequelabs/axe-core/issues/2966#issuecomment-852519866

Do you think showing some information in LH about that Accessibility audit related to color contrast can be wrong?

And maybe disable score impact on contrast until we got fix from Axe?

**Comment by davodavodavo3** [comment]
There is 15 color contrast issues marked as need `fix`

**Comment by patrickhulce** [comment]
> Do you think showing some information in LH about that Accessibility audit related to color contrast can be wrong?
And maybe disable score impact on contrast until we got fix from Axe?

We don't do this for bugs. There are always corner case bugs for any automated check :)

We'll happily update to the latest version when a fix is available though 👍

**Comment by davodavodavo3** [comment]
Hi @patrickhulce there is a new Release 4.2.3 of AXE where the contrast issues have been fixed(as they say).

So maybe it's the time for updating Lighthouse

**Comment by patrickhulce** [comment]
Thanks @davodavodavo3! We've already updated to 4.2.3 so we'll consider this closed then :)

https://github.com/GoogleChrome/lighthouse/blob/175619ae9efffbce73f2d66d6045f0ea851f1296/package.json#L164
